ARGO/src/AiSystem.cpp

```diff
@@ -58,3 +65,46 @@ void AiSystem::simpleMoveAi(TransformComponent* t_posComp)
 		t_posComp->m_movingUp = false;
 	}
 }
+
+void AiSystem::meleeAI(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	//The Only Possible States Available to Melee Enemies. Use to limit Behaviours
+	switch (t_aiComponent->getStates()) {
+	case eSleeping:
+		sleep(t_posComp, t_aiComponent);
+		break;
+	case eWander:
+		wander(t_posComp,t_aiComponent);
+	}
+}
+
+void AiSystem::rangedAI(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	//The Only Possible States Available to Ranged Enemies. Use to limit Behaviours
+	switch (t_aiComponent->getStates()) {
+	case eSleeping:
+		sleep(t_posComp, t_aiComponent);
+		break;
+	case eWander:
+		wander(t_posComp, t_aiComponent);
+	}
+}
+
+void AiSystem::wander(TransformComponent* t_posComp, AiComponent* t_aiComponent)
+{
+	float tempAdjuster = (((rand() % 20 + 1)));//Gives a number between 1 and 20
+	tempAdjuster -= 10; //-10 to give a range of -9 and 10,
+	tempAdjuster /= 10;  //divide by 10 to give a range of -0.9 and 1
+	t_posComp->setRotation(t_posComp->getRotation() + (t_aiComponent->getMaxRotation() * tempAdjuster)); //Randomly changes current angle by the max which is scaled by the random scalar
+	glm::vec2 tempvelocity = glm::vec2(glm::cos(t_posComp->getRotation() * M_PI / 180), sin(t_posComp->getRotation() * M_PI / 180));//Generates a unit vector in the given angle.
+	tempvelocity *= glm::length(t_aiComponent->getMaxSpeed());//Scales it by the length of the max speed.
+	auto tempPosition = t_posComp->getPos();
+	tempPosition.x += tempvelocity.x;
+	tempPosition.y += tempvelocity.y;
+	t_posComp->setPos(tempPosition.x, tempPosition.y);//Updates Position
```

Comment:
You can do these 4 lines in 1:
t_posComp->setPos(t_posComp->getPos() + tempVelocity)

Reply:
if setPos takes only two floats, change to take/overload it to take a vector2